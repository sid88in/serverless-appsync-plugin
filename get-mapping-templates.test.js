const getMappingTemplates = require('./get-mapping-templates');

test('Returns valid mapping templates conf', async () => {
  expect(await getMappingTemplates({
    schema: './example/schema.graphql',
    resolvedConfig: {
      mappingTemplatesLocation: '__test__mapping-templates',
      mappingTemplates: [],
    },
    dataSources: [
      {
        name: 'Lambda',
        description: 'Lambda DataSource',
        autogeneration: false,
        fields: ['meInfo', 'deleteTweet'],
      },
    ],
  })).toMatchSnapshot();
});
