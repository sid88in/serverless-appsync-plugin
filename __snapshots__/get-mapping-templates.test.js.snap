// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Returns valid mapping templates conf 1`] = `
Array [
  Object {
    "dataSource": "Lambda",
    "field": "meInfo",
    "request": "meInfo-request-mapping-template.txt",
    "response": "response-mapping-template.txt",
    "type": "Query",
  },
  Object {
    "dataSource": "Lambda",
    "field": "deleteTweet",
    "request": "deleteTweet-request-mapping-template.txt",
    "response": "response-mapping-template.txt",
    "type": "Mutation",
  },
]
`;
